src/modules/background/background.ts

```diff
@@ -150,34 +166,63 @@ export interface NavigationCommittedParams {
 }
 
 const injectedTabs = new Set();
+let streamingServiceEntries: [StreamingServiceId, StreamingServiceValue][] = [];
 let streamingServiceScripts: browser.runtime.Manifest['content_scripts'] | null = null;
 
 const init = async () => {
 	Shared.pageType = 'background';
-	await BrowserStorage.sync();
-	const storage = await BrowserStorage.get('options');
-	if (storage.options?.allowRollbar) {
+	await BrowserStorage.init();
+	if (BrowserStorage.options.allowRollbar) {
 		Errors.startRollbar();
 	}
 	browser.tabs.onRemoved.addListener((tabId) => void onTabRemoved(tabId));
 	browser.storage.onChanged.addListener(onStorageChanged);
-	if (storage.options?.streamingServices) {
-		const scrobblerEnabled = (Object.entries(storage.options.streamingServices) as [
-			StreamingServiceId,
-			boolean
-		][]).some(
-			([streamingServiceId, value]) => value && streamingServices[streamingServiceId].hasScrobbler
-		);
-		if (scrobblerEnabled) {
-			addWebNavigationListener(storage.options);
-		}
+	streamingServiceEntries = Object.entries(BrowserStorage.options.streamingServices) as [
+		StreamingServiceId,
+		StreamingServiceValue
+	][];
+	const scrobblerEnabled = streamingServiceEntries.some(
+		([streamingServiceId, value]) =>
+			streamingServices[streamingServiceId].hasScrobbler && value.scrobble
+	);
+	if (scrobblerEnabled) {
+		addWebNavigationListener(BrowserStorage.options);
+
+		void checkServicesToSync();
```

Comment:
I don't see why scrobbling should be a requirement for auto-sync

Reply:
Oh that's definitely in the wrong place, good catch!